Boltdir/site-modules/releng_tasks/files/set_travis_env_vars.rb

```diff
@@ -0,0 +1,220 @@
+        @logger.info  "  == Create env_var '#{env_var_name}'"
+        if @noop
+          @logger.info '  -- NOOP: (skipping action)'
+          next
+        end
+        travis_http("#{@travis_api}/repo/#{repo['id']}/env_vars", body: body)
+      else
+        env_var_id = existing_env_vars.first['id']
+        @logger.info "  == Update env_var '#{env_var_name}' (#{env_var_id})"
+        if @noop
+          @logger.info '  -- NOOP: (skipping action)'
+          next
+        end
+        travis_http("#{@travis_api}/repo/#{repo['id']}/env_var/#{env_var_id}",
+                    http_request_type: Net::HTTP::Patch,
+                    body: body)
+      end
+    end
+  end
+
+  def delete_env_var(env_var_name)
+    each_org_repo do |repo|
+      data = travis_http("#{@travis_api}/repo/#{repo['id']}/env_vars")
+      env_vars = data['env_vars']
+
+      existing_env_vars = env_vars.select { |x| x['name'] == env_var_name }
+      if existing_env_vars.empty?
+        @logger.warn "  !! WARNING: env_var '#{env_var_name}' not found"
+      else
+        env_var_id = existing_env_vars.first['id']
+        @logger.info "  == Delete env_var '#{env_var_name}' (#{env_var_id})"
+        if @noop
+          @logger.info '  -- NOOP: (skipping action)'
+          next
+        end
+        travis_http("#{@travis_api}/repo/#{repo['id']}/env_var/#{env_var_id}",
+                    http_request_type: Net::HTTP::Delete)
+      end
+    end
+  end
+
+  def TravisCIOrgEnvSetter.run(options)
+    options['variable']     || raise(ArgumentError,'VARIABLE is required')
+    case options['action']
+    when 'set'
+      options['value']      || raise(ArgumentError,'A VALUE is required to set an env var')
+    end
+    options['travis_token'] || raise('TRAVIS_TOKEN is not set')
+    options['org']          || raise(ArgumentError, 'ORG is required')
+
+    travis_ci_org = TravisCIOrgEnvSetter.new(
+      options['travis_token'],
+      options['org'],
+      options['logdest'] || nil,
+      options['noop'] || false,
+      options['repo_filter'] || nil
+    )
+
+    case options['action']
+    when 'set'
+      travis_ci_org.set_env_var(
+        options['variable'],
+        options['value'],
+        options['public'] || false
+      )
+    when 'delete'
+      travis_ci_org.delete_env_var(options['variable'])
+    end
+
+  end
+end
+
+if __FILE__ == $PROGRAM_NAME
+  # Parse command line
+  options = { 'action' => 'set', 'public' => false, 'noop' => false }
+
+  perma_opts = ''
+  opt_parser = OptionParser.new do |opts|
+    opts.banner = "== #{File.basename($0)} [options]"
+    opts.separator <<-HELP_MSG.gsub(/^ {4}/, '')
+
+      Set a Travis CI environment variable across all of an organization's
+      repositories
+
+      Usage:
+
+        TRAVIS_TOKEN=<TOKEN> #{File.basename($PROGRAM_NAME)} [options] ORG VARIABLE [VALUE]
+
+      Note:
+
+        The environment variable TRAVIS_TOKEN must contain your Travis CI token
+    HELP_MSG
+
+    opts.separator ''
```

Comment:
Why isn't the filter exposed?  For the simp org, wouldn't we want the filter for forked projects we don't own?

Reply:
@lnemsick-simp I *think* that's taken care of by the hard-coded `owner` in https://github.com/simp/releng-misc/pull/4/files/4917d4ff4a5ddaa00b4523cfeb6d9b13ef58a618#diff-c6b37d254630aaed6f31fd13cfd52ed4R59

Reply:
@trevor-vaughan  If you execute this file as is, there is no way to set `repo_filter` from a command line option.  The options parser does not support setting it.  This is distinct from `tasks/set_travis_env_vars.rb`, which allows you to set it with options read in from a JSON file.

Reply:
@op-ct Did you want to do anything with this?